File: web/src/model/victron_controller/dashboard/WeatherSocDay.ts
// @ts-nocheck
import {BaboonCodecContext, BaboonBinWriter, BinTools, BaboonBinReader, Lazy} from '../../BaboonSharedRuntime'

export enum WeatherSocDay {
    Today = "Today",
    Tomorrow = "Tomorrow"
}

export const WeatherSocDay_values: ReadonlyArray<WeatherSocDay> = [
    WeatherSocDay.Today,
    WeatherSocDay.Tomorrow
] as const;

export function WeatherSocDay_parse(s: string): WeatherSocDay {
    const found = WeatherSocDay_values.find(v => v === s);
    if (found === undefined) {
        throw new Error("Unknown WeatherSocDay variant: " + s);
    }
    return found;
}

export class WeatherSocDay_UEBACodec {
    public encode(ctx: BaboonCodecContext, value: WeatherSocDay, writer: BaboonBinWriter): unknown {
        if (this !== WeatherSocDay_UEBACodec.lazyInstance.value) {
          return WeatherSocDay_UEBACodec.lazyInstance.value.encode(ctx, value, writer)
        }
    
        switch (value) {
            case "Today": BinTools.writeByte(writer, 0); break;
                case "Tomorrow": BinTools.writeByte(writer, 1); break;
            default: throw new Error("Unknown enum variant: " + value);
        }
    }
    
    public decode(ctx: BaboonCodecContext, reader: BaboonBinReader): WeatherSocDay {
        if (this !== WeatherSocDay_UEBACodec .lazyInstance.value) {
            return WeatherSocDay_UEBACodec.lazyInstance.value.decode(ctx, reader)
        }
    
        const tag = BinTools.readByte(reader);
        switch (tag) {
            case 0: return "Today" as WeatherSocDay;
                case 1: return "Tomorrow" as WeatherSocDay;
            default: throw new Error("Unknown enum variant tag: " + tag);
        }
    }

    public static readonly BaboonDomainVersion = '0.3.0'
    public baboonDomainVersion() {
        return WeatherSocDay_UEBACodec.BaboonDomainVersion
    }
    public static readonly BaboonDomainIdentifier = 'victron_controller.dashboard'
    public baboonDomainIdentifier() {
        return WeatherSocDay_UEBACodec.BaboonDomainIdentifier
    }
    public static readonly BaboonTypeIdentifier = 'victron_controller.dashboard/:#WeatherSocDay'
    public baboonTypeIdentifier() {
        return WeatherSocDay_UEBACodec.BaboonTypeIdentifier
    }

    protected static lazyInstance = new Lazy(() => new WeatherSocDay_UEBACodec())
    public static get instance(): WeatherSocDay_UEBACodec {
        return WeatherSocDay_UEBACodec.lazyInstance.value
    }
}

File: crates/dashboard-model/src/victron_controller/dashboard/weather_soc_day.rs


#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, serde::Serialize, serde::Deserialize)]
pub enum WeatherSocDay {
    Today,
    Tomorrow,
}

impl WeatherSocDay {
    pub fn parse(s: &str) -> Result<Self, String> {
        match s {
            "Today" => Ok(WeatherSocDay::Today),
            "Tomorrow" => Ok(WeatherSocDay::Tomorrow),
            _ => Err(format!("Unknown variant: {}", s)),
        }
    }

    pub fn all() -> Vec<WeatherSocDay> {
        vec![
            WeatherSocDay::Today,
            WeatherSocDay::Tomorrow,
        ]
    }
}

impl std::fmt::Display for WeatherSocDay {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            WeatherSocDay::Today => write!(f, "Today"),
            WeatherSocDay::Tomorrow => write!(f, "Tomorrow"),
        }
    }
}

impl std::str::FromStr for WeatherSocDay {
    type Err = String;
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        Self::parse(s)
    }
}

impl crate::baboon_runtime::BaboonBinCodecIndexed for WeatherSocDay {
    fn index_elements_count(_ctx: &crate::baboon_runtime::BaboonCodecContext) -> u16 {
        0
    }
}

impl crate::baboon_runtime::BaboonBinEncode for WeatherSocDay {
    fn encode_ueba(&self, _ctx: &crate::baboon_runtime::BaboonCodecContext, writer: &mut dyn std::io::Write) -> std::io::Result<()> {
        match self {
            WeatherSocDay::Today => crate::baboon_runtime::bin_tools::write_byte(writer, 0)?,
            WeatherSocDay::Tomorrow => crate::baboon_runtime::bin_tools::write_byte(writer, 1)?,
        }
        Ok(())
    }
}

impl crate::baboon_runtime::BaboonBinDecode for WeatherSocDay {
    fn decode_ueba(_ctx: &crate::baboon_runtime::BaboonCodecContext, reader: &mut dyn std::io::Read) -> Result<Self, Box<dyn std::error::Error>> {
        let tag = crate::baboon_runtime::bin_tools::read_byte(reader)?;
        match tag {
            0 => Ok(WeatherSocDay::Today),
            1 => Ok(WeatherSocDay::Tomorrow),
            _ => Err(format!("Unknown enum variant tag: {}", tag).into()),
        }
    }
}